DEVELOPING.md

```diff
@@ -1,75 +1,241 @@
+Build & Test Toolbox IDE
+------------------------
+*Note: these instructions are likely to fail and must be fixed; see [this issue](https://github.com/tlaplus/tlaplus/issues/851).*
 
-Make sure that you're using Java 11+. If you're using a lower version you'll see compilation errors.
+Install the following dependencies to your path:
+ * [OpenJDK](https://openjdk.org/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+ * [Apache Maven](https://maven.apache.org/) version 3.9.7+
 
-You can build TLC from a fresh clone of this repository+:
+Clone this repo & open a shell in its root, then run:
 ```bash
-$ git clone https://github.com/tlaplus.git
-$ cd tlaplus/tlatools/org.lamport.tlatools
-$ ant -f customBuild.xml info clean compile compile-test dist
+mvn install -Dmaven.test.skip=true # Builds Toolbox
+mvn verify # Runs tests
+```
+
+The Toolbox distributables will be output in the `toolbox/org.lamport.tla.toolbox.product.product/target/products` directory.
+
+Developing with the Eclipse IDE
+-------------------------------
+
+You are free to use your own editor, but this project is largely developed in the [Eclipse](https://www.eclipse.org/) integrated development environment (IDE).
+
+### Developing the TLA⁺ Tools in Eclipse
+
+To develop the TLA⁺ Tools in Eclipse, use Eclipse to open the `tlatools/org.lamport.tlatools` project directory.
+Within that directory, the Eclipse project configuration lives in the `.classpath` and `.project` files.
+The `.classpath` file is also consumed by the Toolbox IDE build, so some local modifications are required before Eclipse can fully build & test the project.
+Copy & paste the following into a file called `classpath.patch` within your repo:
+```diff
+diff --git a/tlatools/org.lamport.tlatools/.classpath b/tlatools/org.lamport.tlatools/.classpath
+index c4c89e181..2caf43a97 100644
+--- a/tlatools/org.lamport.tlatools/.classpath
++++ b/tlatools/org.lamport.tlatools/.classpath
+@@ -26,11 +26,15 @@
+ 	<classpathentry kind="lib" path="lib/jline/jline-builtins-3.25.0.jar"/>
+ 	<classpathentry kind="lib" path="lib/jline/jline-reader-3.25.0.jar"/>
+ 	<classpathentry kind="lib" path="lib/jline/jline-terminal-3.25.0.jar"/>
+-	<classpathentry kind="lib" path="lib/jpf.jar"/>
+ 	<classpathentry kind="lib" path="lib/jmh/jmh-core-1.21.jar"/>
+ 	<classpathentry kind="lib" path="lib/jmh/commons-math3-3.2.jar"/>
+ 	<classpathentry kind="lib" path="lib/gson/gson-2.8.6.jar"/>
+ 	<classpathentry kind="lib" path="lib/lsp/org.eclipse.lsp4j.debug_0.21.1.v20230829-0012.jar"/>
+ 	<classpathentry kind="lib" path="lib/lsp/org.eclipse.lsp4j.jsonrpc_0.21.1.v20230829-0012.jar"/>
++	<classpathentry kind="lib" path="lib/junit-4.12.jar"/>
++	<classpathentry kind="lib" path="lib/jpf.jar"/>
++	<classpathentry kind="lib" path="lib/easymock-3.3.1.jar"/>
```

Comment:
The tools use Easymock 2.4.0.

Reply:
the `tlatools/org.lamport.tlatools/lib` directory only contains `easymock-3.3.1.jar`.

Reply:
With https://github.com/tlaplus/tlaplus/pull/953 merged, I can now import the `org.lamport.tlatools` project into a recent Eclipse IDE, and it compiles successfully out of the box.

Reply:
That's great! I can confirm the tools build successfully now. However, the unit tests still do not build in Eclipse without the addition of `lib/junit-4.12.jar` to the `.classpath` file, and they must be placed above `lib/jpf.jar` (due to some class name collisions). The unit tests will also fail to run unless `lib/hamcrest-core-1.3.jar` is included in `.classpath`.

Reply:
Please check if your Eclipse IDE comes with (OSGi) Plugin Development Extension (PDE), because junit and hamcrest are provided by its plugin libraries.

<img width="1037" alt="image" src="https://github.com/tlaplus/tlaplus/assets/88777/4c66b28d-f2e5-46e7-89a1-c66a0f716324">

Reply:
Oh excellent, I found that the [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2024-06/r/eclipse-ide-rcp-and-rap-developers) edition comes with that plugin - I installed that one instead of the one I have now (I think just the basic "Eclipse for Java" edition) and now I can build & run the tests! I will add this to the instructions.